hydroflow/src/scheduled/net/network_vertex.rs

```diff
@@ -18,6 +21,100 @@ pub type ReceiverId = u32;
 pub type OutboundMessage = (ReceiverId, Message);
 
 impl Hydroflow {
+    // TODO(justin): this needs to return a result/get rid of all the unwraps, I
+    // guess we need a HydroflowError?
+    pub async fn listen_tcp(
+        &mut self,
+    ) -> (
+        u16,
+        OutputPort<VecHandoff<(ReceiverId, Message)>>,
+        OutputPort<VecHandoff<ReceiverId>>,
+        InputPort<VecHandoff<(u32, Message)>>,
+    ) {
+        let listener = TcpListener::bind("localhost:0").await.unwrap();
+        let port = listener.local_addr().unwrap().port();
+
+        let (mut conns, egress_edge) = self.add_egress_vertex();
+        // TODO(justin): figure out an appropriate buffer here.
+        let (incoming_send, incoming_messages) = futures::channel::mpsc::channel(1024);
+        let (mut incoming_conns, conns_recv) = futures::channel::mpsc::channel(1024);
+
+        // Listen to incoming connections and spawn a tokio task for each one,
+        // which feeds into the channel.
+        tokio::spawn(async move {
```

Comment:
Loose hidden `tokio::spawn` is suspicious but I'm fine with a TODO here for fixing it

Reply:
What's suspicious about it? Don't we have to spawn to continually grab incoming connections? Do you mean we don't have a way to shutdown the server?

Reply:
Yeah, no way to shut down or handle errors, hidden side effects of calling a method

Reply:
tokio::spawn returns a `JoinHandle`, so should be able to track. https://tokio.rs/tokio/tutorial/spawning